test/test_constraints.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+import numpy as np
+from sklearn.utils import shuffle
+from metric_learn.constraints import Constraints
+
+SEED = 42
+
+
+def gen_labels_for_chunks(num_chunks, chunk_size,
+                          n_classes=10, n_unknown_labels=5):
+  """Generates num_chunks*chunk_size labels that split in num_chunks chunks,
+  that are homogeneous in the label."""
+  assert min(num_chunks, chunk_size) > 0
+  classes = shuffle(np.arange(n_classes), random_state=SEED)
+  n_per_class = chunk_size * (num_chunks // n_classes)
+  n_maj_class = n_per_class + chunk_size * num_chunks - n_per_class * n_classes
+
+  first_labels = classes[0] * np.ones(n_maj_class, dtype=int)
+  remaining_labels = np.concatenate([k * np.ones(n_per_class, dtype=int)
+                                     for k in classes[1:]])
+  unknown_labels = -1 * np.ones(n_unknown_labels, dtype=int)
+
+  labels = np.concatenate([first_labels, remaining_labels, unknown_labels])
+  return shuffle(labels, random_state=SEED)
+
+
+@pytest.mark.parametrize('num_chunks, chunk_size', [(11, 5), (115, 12)])
+def test_chunk_case_exact_num_points(num_chunks, chunk_size,
+                                     n_classes=10, n_unknown_labels=5):
+  """Checks that the chunk generation works well with just enough points."""
+  labels = gen_labels_for_chunks(num_chunks, chunk_size,
+                                 n_classes=n_classes,
+                                 n_unknown_labels=n_unknown_labels)
+  constraints = Constraints(labels)
+  chunks = constraints.chunks(num_chunks=num_chunks, chunk_size=chunk_size,
+                              random_state=SEED)
+  return chunks
```

Comment:
Instead of simply returning `chunks`, it would be nice to test that `chunks` contains `num_chunks` chunks, each composed of `chunk_size` points

Reply:
I will add it.

A remark, I expected chunks to map the *datapoint index* (input order) to the chunk number, but chunks removes -1 instances (unknown labels, according to line 21), which means that one has to consider *datapoint indexes* when removing -1's. But it must be taken into account in the implementation of RCA.

Reply:
Can you elaborate on this? Not sure I understand what you mean.

Do we test at all `RCA_Supervised` in the presence of unlabeled points? At first sight, it looks like it might fail due to unlabeled points being removed from the chunk array as pointed out by
@RobinVogel. @wdevazelhes, any comment?

Reply:
I'll just add an example to make it clearer. Below is a comparison in the presence of unknown labels between the output that I expected from `Constraints.chunks`, i.e. `expected_chunk` and the one that I obtain from `Constraints.chunks`, i.e. `chunks`:
```python
In [1]: from metric_learn.constraints import Constraints                                                                                                                                                        

In [2]: partial_labels = [1, 2, 2, 1, -1, 3, 3]                                                        

In [3]: cons = Constraints(partial_labels)                                                             

In [4]: cons.chunks(num_chunks=3, chunk_size=2)                                                        
Out[4]: array([0, 1, 1, 0, 2, 2])

In [5]: chunks = cons.chunks(num_chunks=3, chunk_size=2)                                               

In [6]: len(chunks), len(partial_labels)                                                               
Out[6]: (6, 7)

In [7]: expected_chunk = [0, 1, 1, 0, -1, 2, 2]
```

Reply:
Thanks. And this is indeed a problem for `RCA_Supervised`, see the issue #260 that I just added. I have also added another issue (#261) to make clear in the doc that `-1` is interpreted as unlabeled. You're very welcome to address those @RobinVogel while you have everything well in mind :-)